robot/packages/utils.go

```diff
@@ -0,0 +1,96 @@
+package packages
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/logging"
+)
+
+// downloadCallback is the function signature that gets passed to downloadPackage.
+type downloadCallback func(ctx context.Context, url, dstPath string) (contentType string, err error)
+
+func downloadPackage(ctx context.Context, logger logging.Logger, packagesDir, url string, p config.PackageConfig,
+	nonEmptyPaths []string, downloadFn downloadCallback,
+) error {
+	if dataDir := p.LocalDataDirectory(packagesDir); dirExists(dataDir) {
+		if checkNonemptyPaths(p.Name, logger, nonEmptyPaths) {
```

Comment:
`checkNonemptyPaths` is defined in the `cloud_package_manager` 

I think we should bring it out here too.

Reply:
done in 4e854649